usaspending_api/search/v2/views/spending_by_category_views/base_spending_by_industry_codes.py

```diff
@@ -0,0 +1,75 @@
+import json
+from abc import ABCMeta
+from decimal import Decimal
+from enum import Enum
+from typing import List
+
+from django.db.models import QuerySet, F
+
+from usaspending_api.search.helpers.spending_by_category_helpers import (
+    fetch_naics_description_from_code,
+    fetch_psc_description_by_code,
+    fetch_cfda_id_title_by_number,
+)
+from usaspending_api.search.v2.views.spending_by_category_views.base_spending_by_category import (
+    BaseSpendingByCategoryViewSet,
+)
+
+
+class IndustryCodeType(Enum):
+    CFDA = "cfda_number"
+    PSC = "product_or_service_code"
+    NAICS = "naics_code"
+
+
+class BaseIndustryCodeViewSet(BaseSpendingByCategoryViewSet, metaclass=ABCMeta):
```

Comment:
Stylistic thing, but one I'm fond of: when creating an abstract class, name it with `Abstract`, e.g. `AbstractIndustryCodeViewSet`. It implies that it's base, and also alerts people that it's not instantiable. If that's too verbose, you could use `AIndustryCodeViewSet` or `ACFDAViewSet` (of course we use uppercase for acronyms on this project 😐 ... so that one looks bad)